CodeEdit/Features/CEWorkspace/Models/CEWorkspaceFile.swift

```diff
@@ -234,7 +234,8 @@ final class CEWorkspaceFile: Codable, Comparable, Hashable, Identifiable, Editor
     // MARK: Hashable
 
     func hash(into hasher: inout Hasher) {
-        hasher.combine(fileIdentifier)
+// TODO: Causing crashes, needs a better solution and should be fixed before merging.
```

Comment:
@thecoolwinter @Wouter01 we discussed this on Discord. To recap this was causing crashes in the changes tab of the source control navigator.

![image](https://github.com/CodeEditApp/CodeEdit/assets/806104/0ef86880-619b-44a7-9cad-fac3ea199cf1)

![image](https://github.com/CodeEditApp/CodeEdit/assets/806104/01f9c255-8aaa-4c93-a0f2-ed7af44328d6)

This seemed to be the code I added when I first started noticing issues...

```swift
struct SourceControlNavigatorChangesView: View {
    @ObservedObject var sourceControlManager: SourceControlManager

    @State var selection = Set<CEWorkspaceFile>()

    // ...
```

```swift
List(sourceControlManager.changedFiles, id: \.self, selection: $selection) { file in
    SourceControlNavigatorChangedFileView(
        sourceControlManager: sourceControlManager,
        changedFile: file
    )
    .listRowSeparator(.hidden)
}
```

https://github.com/CodeEditApp/CodeEdit/assets/806104/97aee348-b4b8-4d8a-a62c-6b9baca8452d

![image](https://github.com/CodeEditApp/CodeEdit/assets/806104/2a222aa0-c31a-4f18-90f0-3233406f654e)

Reply:
This doesn't seem to cause problems from what I can tell. I am fine merging as is unless y'all think this is a problem.

Reply:
It shouldn't be, if the file is moved we create a new `CEWorkspaceFile` afaik. Either way it's not being mutated right now anywhere in the project and `id` is already based on `url`.

Reply:
Okay, we can do that then. I know if a file in an open tab in VS Code is moved or deleted, it crosses out the tabs label indicating that the open file has been deleted. The tab is then treated as a temporary untitled tab (even thought the tabs label remains the same) and the user can make changes. When the user saves it will save the file where it originally was. The user also has the option to save as to relocate the file.

Reply:
Okay, I've made the requested changes. Double check to see if that is what you wanted.